doc/en/reference.rst

```diff
@@ -1561,6 +1561,18 @@ passed multiple times. The expected format is ``name=value``. For example::
    See :ref:`change naming conventions` for more detailed examples.
 
 
+.. confval:: require_plugins
+
+   A space separated list of plugins that must be present for pytest to run.
+   If any one of the plugins is not found, emit a warning.
+   If pytest is run with ``--strict-config`` exceptions are raised in place of warnings.
+
+   .. code-block:: ini
+
+       [pytest]
+       require_plugins = html xdist
```

Comment:
@RonnyPfannschmidt @nicoddemus what if we just checked for both `<plugin>` and `pytest-<plugin>` ( e.g.: `html` and `pytest-html` ) as suggested in the beginning of the comment. To @nicoddemus' point, there is no standard way to define plugins, and once all the plugins are loaded we store them in the way I wrote up with this pr ( `pytest-html` becomes `html` ).

I think looking for both would be valid, since one is the internal representation, and the other is the user friendly name, and it would be simpler than trying to user the logic in `terminal.py` which we'd probably have to groom a little for this use case

Reply:
We shouldn't guess, the giben names should be python distribution names

With module names also being accepted

There should be a way to add version constraints

Reply:
@RonnyPfannschmidt - so you mean we should prefer `pytest-html` over `html`

Agreed on the ability to add version constraints. Can we add this in a follow up PR to not bloat this one?

Reply:
@RonnyPfannschmidt - so you mean we should prefer `pytest-html` over `html`

Agreed on the ability to add version constraints. Can we add this in a follow up PR to not bloat this one?

Reply:
Also, @RonnyPfannschmidt @nicoddemus is it a requirement that plugins for pytest start with `pytest-`? It seems like yes from the documentation, but I am not sure

Reply:
Sure

Reply:
@nicoddemus - thanks for sharing the snippet in `pytest/src/_pytest/terminal.py`, we  can definitely use that as a solution here